python/nano/src/bigdl/nano/pytorch/runtime_binding/openvino_inference.py

```diff
@@ -0,0 +1,183 @@
+
+def forward_batch_start(*args):
+    ort_inputs = []
+    for ort_input_item in args:
+        if isinstance(ort_input_item, torch.Tensor):
+            ort_input_item = ort_input_item.numpy()
+        ort_inputs.append(ort_input_item)
+    return ort_inputs
+
+
+def forward_batch_end(*outputs):
+    return torch.from_numpy(outputs[0])
+
+
+def forward(self, *args):
+    args = forward_batch_start(*args)
+    outputs = _forward_openvino(self, *args)
+    return forward_batch_end(*outputs.values())
+
+
+def _forward_openvino(self, *args):
+    inputs = dict(zip(self.ir_model.input_info, args))
+    return self.ir_model.infer(inputs)
+
+
+def eval_openvino(self, input_sample=None, file_path="model.xml", quantize=False, **kwargs):
+    '''
+    This method change the `forward` method to an openvino backed forwarding.
+
+    >>> model.eval_openvino(quantize=True/False)
+    >>> pred = model(x)  # openvino forwarding
+    >>> model.exit_openvino()
+
+    :param input_sample: (optional) a torch dataloader, torch.Tensor or a
+           list of them for the model tracing.
+    :param file_path: (optional) The path to save openvino model file.
+    :param quantize: Bool, state if we need to use quantized openvino session.
+    :param **kwargs: (optional) will be passed to torch.onnx.export function.
+    '''
+    # change to eval mode
+    self.eval()
+    if not self.ir_up_to_date:
+        # get input_sample
+        if isinstance(input_sample, DataLoader):
+            input_sample = tuple(next(iter(input_sample))[:-1])
+        elif input_sample is None:
+            if self.example_input_array:
+                input_sample = self.example_input_array
+            if self.trainer is None:
+                raise RuntimeError("You must specify an input_sample or call `Trainer.fit` "
+                                   "on the model first to use `eval_openvino`")
+            if self.train_dataloader:
+                input_sample = tuple(next(iter(self.train_dataloader()))[:-1])
+        assert input_sample is not None,\
+            "You must state an input_sample or fit on the model to use `eval_onnx`."
+        _build_ir_model(self, input_sample=input_sample, file_path=file_path, **kwargs)
+    self._torch_train = self.train
+    self._torch_eval = self.eval
+    self.train = partial(train, self)
+    self.forward = partial(forward, self)
+
+
+def exit_openvino(self):
+    self.train = self._torch_train
+    self.forward = self._torch_forward
+
+
+def bind_openvino_methods(pl_model: LightningModule):
+
+    if set(OPENVINO_BINDED_COMPONENTS) <= set(dir(pl_model)):
+        return pl_model
+
+    # check conflicts
+    for component in OPENVINO_BINDED_COMPONENTS:
+        if component in dir(pl_model):
+            warnings.warn(f"{component} method/property will be replaced. You may rename your"
+                          " customized attributes or methods and call `Trainer.compile again `"
+                          "to avoid being overwrite.")
```

Comment:
overwritten

Reply:
Removed this part since we just need to bind openvino method only once.